sdk/confidentialledger/Azure.Security.CodeTransparency/tests/CcfReceiptVerifierTests.cs

```diff
@@ -0,0 +1,231 @@
+#if NETFRAMEWORK
+            Assert.True(ex.Message.Contains("One or more errors occurred"));
+#else
+            Assert.That(ex.Message, Contains.Substring("VerifyReceipt: signature invalid"));
+#endif
+        }
+
+        [Test]
+        public void Verify_separate_receipt_example2_success_Test()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-11-10.receipt.2023-01-23.cbor");
+            byte[] coseSign1Bytes = readFileBytes("sbom.descriptor.2022-11-10.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes, coseSign1Bytes);
+        }
+
+        [Test]
+        public void Verify_separate_receipt_example3_success_Test()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.receipt.2023-01-23.cbor");
+            byte[] coseSign1Bytes = readFileBytes("sbom.descriptor.2022-12-10.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes, coseSign1Bytes);
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_example_success()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.embedded.2023-02-13.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes);
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_is_missing_did_issuer()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.embedded.2023-02-13.cose");
+            var didDocBytes = readFileBytes("service.2022-11.did.json");
+            var didDoc = DidDocument.DeserializeDidDocument(JsonDocument.Parse(didDocBytes).RootElement);
+            var ex = Assert.Throws<AggregateException>(() => CcfReceiptVerifier.RunVerification(receiptBytes, null, didRef => didDoc));
+#if NETFRAMEWORK
+            Assert.True(ex.Message.Contains("One or more errors occurred"));
+#else
+            Assert.That(ex.Message, Contains.Substring("invalid did:web issuer: null"));
+#endif
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_with_did_issuer()
+        {
+            #region Snippet:CodeTransparencyVerification
+            #region Snippet:CodeTransparencySample2_VerifyEntryWithEmbeddedReceipt
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.embedded.did.2023-02-13.cose");
+            var didDocBytes = readFileBytes("service.2023-03.did.json");
+            var didDoc = DidDocument.DeserializeDidDocument(JsonDocument.Parse(didDocBytes).RootElement);
+            CcfReceiptVerifier.RunVerification(receiptBytes, null, didRef => {
+                Assert.AreEqual("https://preview-test.scitt.azure.net/.well-known/did.json", didRef.DidDocUrl.ToString());
```

Comment:
```suggestion
#if !SNIPPET
                Assert.AreEqual("https://preview-test.scitt.azure.net/.well-known/did.json", didRef.DidDocUrl.ToString());
#endif
```

Reply:
This is an example of what you'd have to do to scrub the non-sample bits from the snippet.